lib/spack/llnl/util/filesystem.py

```diff
@@ -1180,6 +1178,26 @@ def remove_if_dead_link(path):
         os.unlink(path)
 
 
+class CallingVisitor(object):
+    def __init__(self, callable):
+        self.caller = callable
+
+    def before_visit_dir(self, root, rel_path, depth):
+        return True
+
+    def after_visit_dir(self, root, rel_path, depth):
+        self.caller(os.path.join(root, rel_path))
+
+    def before_visit_symlinked_dir(self, root, rel_path, depth):
+        return False
+
+    def after_visit_symlink(self, root, rel_path, depth):
+        self.caller(os.path.join(root, rel_path))
```

Comment:
I think this is a typo? Maybe we should just implement a dummy interface class for the visitor, so that you don't have to implement `after_visit_symlinked_dir` when you don't need it.

Reply:
Yes.
I agree, I was surprised when I didn''t find one already, I'll create a `GenericVisitor` class to encompass that